hal/qspi_api.h

```diff
@@ -141,27 +141,33 @@ qspi_status_t qspi_free(qspi_t *obj);
  */
 qspi_status_t qspi_frequency(qspi_t *obj, int hz);
 
-/** Send only QSPI command
+/** Send a command and block of data
  *
  * @param obj QSPI object
  * @param command QSPI command
- * @return QSPI_STATUS_OK if command was sent without any error
+ * @param data TX buffer
+ * @param[in,out] in - length TX buffer length in bytes, out - number of bytes written
+ * @return QSPI_STATUS_OK if the data has been succesfully sent
            QSPI_STATUS_INVALID_PARAMETER if invalid parameter found
            QSPI_STATUS_ERROR otherwise
  */
-qspi_status_t qspi_write_command(qspi_t *obj, const qspi_command_t *command);
+qspi_status_t qspi_write(qspi_t *obj, const qspi_command_t *command, const void *data, size_t *length);
 
-/** Send a command and block of data
+/** Send a command (and optionally data) and get the response. Can be used to send/receive device specific commands.
  *
  * @param obj QSPI object
  * @param command QSPI command
- * @param data TX buffer
- * @param[in,out] in - length TX buffer length in bytes, out - number of bytes written
+ * @param tx_data TX buffer
+ * @param tx_length pointer to variable holding TX buffer length
+ * @param rx_data TX buffer
+ * @param rx_length pointer to variable holding TX buffer length
  * @return QSPI_STATUS_OK if the data has been succesfully sent
            QSPI_STATUS_INVALID_PARAMETER if invalid parameter found
            QSPI_STATUS_ERROR otherwise
  */
-qspi_status_t qspi_write(qspi_t *obj, const qspi_command_t *command, const void *data, size_t *length);
+
+qspi_status_t qspi_command_transfer(qspi_t *obj, const qspi_command_t *command, const void *tx_data, size_t tx_size, void *rx_data, size_t rx_size); 
```

Comment:
I think we should keep the name qspi_command_transfer( or something similar ). What it basically provides is a way to send a command specific to target peripheral if needed. This function may not even be implemented by all HALs if it doesnt support it. So if STM HAL doesn't implement sending specific commands, then it can be implemented to return error code.

Reply:
still not sold about the arguments and how it  will be handled in the driver + user space. Lets leave it as it is, and will return to this later.